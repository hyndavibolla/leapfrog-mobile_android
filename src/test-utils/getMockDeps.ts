import { ENV } from '_constants';
import { GeneralModel, MissionModel } from '_models';
import { RUNTIME_ENV } from '_models/general';
import { ICrashlyticsService } from '_services/CrashlyticsService';
import { IHttpService } from '_services/HttpService';
import { IPerformanceTrackingService } from '_services/PerformanceTrackingService';
import { getInitialState } from '_state_mgmt/GlobalState';
import {
  getUser,
  getGame,
  getActivity_1,
  getActivity_2,
  getActivity_3,
  getMissionCategory_1,
  getMissionCategory_2,
  getMissionCategory_3,
  getMissionKeyword_1,
  getMissionKeyword_2,
  getMissionKeyword_3,
  getMission_1,
  getRewardConfig_1,
  getSlideObject_1,
  getSlideObject_2,
  getSlideCategoryIdList,
  getStreak_1,
  getQuestion_1,
  getQuestion_2,
  getQuestion_3,
  getAnswer_1,
  getAnswer_2,
  getAnswer_3,
  getNewOnMaxMessage_1,
  getNewOnMaxMessage_2,
  getNewOnMaxMessage_3,
  getSailthruMessage_1,
  getLinkedCards_1,
  getLinkedCards_2,
  getLinkedCards_3,
  getLinkedCards_4,
  getNewOnMaxMessage_4,
  getGiftCard,
  getGiftCardDetail,
  getGiftCardBalance
} from '_test_utils/entities';
import { createUUID } from '_utils/create-uuid';

export const getMockDeps = (initialStateGetter = getInitialState): GeneralModel.Deps => ({
  apiService: {
    isTokenInvalid: false,
    getDeviceIdAsync: async () => createUUID(),
    isTokenSetCurrentlyReady: () => true,
    setTokenSet: jest.fn(),
    authenticate: jest
      .fn()
      .mockResolvedValue({ accessToken: 'accessToken', refreshToken: 'refreshToken', accessTokenExpiryTime: 'accessTokenExpiryTime', deviceId: 'deviceId' }),
    enrollCard: jest.fn().mockResolvedValue(undefined),
    fetchRefId: jest.fn().mockResolvedValue('test-ref-id'),
    fetchUserProfile: jest.fn().mockResolvedValue(getUser()),
    fetchGameState: jest.fn().mockResolvedValue(getGame()),
    fetchRewardConfig: jest.fn().mockResolvedValue(getRewardConfig_1()),
    fetchActivityHistory: jest.fn().mockResolvedValue([getActivity_1(), getActivity_2(), getActivity_3()]),
    fetchGiftCardList: jest.fn().mockResolvedValue([getGiftCard()]),
    fetchGiftCardDetail: jest.fn().mockResolvedValue(getGiftCardDetail()),
    fetchGiftCardBalance: jest.fn().mockResolvedValue(getGiftCardBalance()),
    fetchWalletStats: jest.fn().mockResolvedValue({
      totalLifetimeEarnedPoints: 15550,
      monthlySummaries: {
        '2020-02': 1500,
        '2020-03': 2000,
        '2020-04': 100,
        '2020-05': 950,
        '2020-06': 1000,
        '2020-07': 1500
      }
    }),
    fetchMissionCategoryList: jest.fn().mockResolvedValue([getMissionCategory_1(), getMissionCategory_2(), getMissionCategory_3()]),
    fetchMissionKeywordList: jest.fn().mockResolvedValue({
      [MissionModel.KeywordType.CATEGORY]: [getMissionKeyword_1().value],
      [MissionModel.KeywordType.BRAND]: [getMissionKeyword_2().value, getMissionKeyword_3().value]
    }),
    fetchMissionList: jest.fn(async ({ listType }) => ({ userId: 'buttonId', missions: [getMission_1()], title: 'mission endpoint title', listType })),
    registerDeviceInfo: jest.fn(),
    registerUserActivity: jest.fn(),
    logout: jest.fn(),
    getSYWOnlineToken: jest.fn().mockResolvedValue('test-token'),
    sendValidationEmail: jest.fn().mockResolvedValue(undefined),
    fetchStreakList: jest.fn().mockResolvedValue([getStreak_1()]),
    fetchPromptList: jest.fn().mockResolvedValue({
      promptQuestions: [getQuestion_1(), getQuestion_2(), getQuestion_3()],
      answeredQuestions: [getAnswer_1(), getAnswer_2(), getAnswer_3()]
    }),
    updatePromptList: jest.fn(),
    unregisterDeviceInfo: jest.fn(),
    fetchLinkedCardsList: jest.fn().mockResolvedValue({ linkedCards: [getLinkedCards_1(), getLinkedCards_2(), getLinkedCards_3(), getLinkedCards_4()] }),
    activateLocalOffer: jest.fn(),
    fetchAppleWalletPass: jest.fn().mockResolvedValue('test-pass'),
    updateCardStatus: jest.fn()
  } as any,
  appService: {
    getDeviceIdAsync: async () => createUUID()
  } as any,
  crashlyticsService: {
    initialize: jest.fn(),
    log: jest.fn(),
    recordError: jest.fn(),
    setCrashlyticsCollectionEnabled: jest.fn()
  } as ICrashlyticsService,
  httpService: {
    fetch: jest.fn()
  } as IHttpService,
  performanceTrackingService: {
    initialize: jest.fn(),
    measuredFetch: jest.fn(),
    setPerformanceCollectionEnabled: jest.fn()
  } as IPerformanceTrackingService,
  raiseService: {
    fetchSlideBrand: jest.fn().mockResolvedValue([getSlideObject_1(), getSlideObject_2()]),
    fetchSlideCategoryIdList: jest.fn().mockResolvedValue(getSlideCategoryIdList())
  } as any,
  nativeHelperService: {
    alert: {
      alert: jest.fn(),
      prompt: jest.fn()
    },
    appState: { addEventListener: jest.fn(), currentState: 'active' },
    storage: {
      get: jest.fn().mockResolvedValue(null),
      set: jest.fn().mockResolvedValue(null),
      remove: jest.fn().mockResolvedValue(null)
    },
    clipboard: {
      getString: jest.fn(),
      setString: jest.fn()
    },
    dimensions: {
      getWindowWidth: jest.fn().mockReturnValue(750),
      getWindowHeight: jest.fn().mockReturnValue(1000),
      isSmallDevice: false
    },
    deviceInfo: {
      getDeviceId: createUUID,
      getUniqueId: createUUID,
      getVersion: () => '0.0',
      getBuildNumber: () => '0.1',
      getDeviceModel: () => 'iPhone 11',
      getLocales: { languageTag: 'en-US' },
      getBaseOs: async () => 'iOS',
      getSystemVersion: () => '11.0',
      getDeviceToken: () => createUUID(),
      getDeviceTokenAsync: async () => createUUID(),
      getDeviceOsType: async () => 'iOS',
      getAppVersion: () => '0.0.0.1',
      getDeviceLanguage: () => 'en-US',
      getDeviceOsVersion: () => '11.0',
      getFCMTokenAsync: async () => createUUID(),
      getTimeZone: () => 'testing-timezone',
      getReadableVersion: () => '1.1.1.1',
      onFCMTokenRefresh: jest.fn()
    },
    netInfo: {
      getState: jest.fn().mockResolvedValue({ isConnected: true }),
      subscribeToConnectionChanges: jest.fn(),
      waitForConnection: jest.fn().mockResolvedValue(undefined)
    },
    linking: {
      openURL: jest.fn(() => Promise.resolve()),
      canOpenURL: jest.fn(),
      checkURLScheme: jest.fn().mockResolvedValue(true),
      getInitialURL: jest.fn(),
      addEventListener: jest.fn()
    },
    buttonSdk: {
      configure: jest.fn(),
      setUserId: jest.fn(),
      clearData: jest.fn(),
      purchaseRequest: jest.fn()
    },
    platform: {
      os: 'default',
      select: jest.fn(config => config?.default)
    },
    fireBaseMessaging: {
      getToken: jest.fn().mockReturnValue(createUUID())
    },
    hash: {
      buildSHA256: jest.fn().mockResolvedValue('hash')
    },
    restart: jest.fn(),
    cookieManager: {
      clearAll: jest.fn()
    },
    base64: {
      encode: jest.fn((...args) => `b64 encoded: ${JSON.stringify(args)}`),
      decode: jest.fn((...args) => `b64 decoded: ${JSON.stringify(args)}`)
    },
    getRuntimeEnv: jest.fn().mockResolvedValue(RUNTIME_ENV.DEV_TEST),
    messaging: {
      hasPermission: jest.fn().mockReturnValue(true),
      requestPermission: jest.fn()
    },
    pushNotificationsAuthStatus: {
      NOT_DETERMINED: -1,
      DENIED: 0,
      AUTHORIZED: 1,
      PROVISIONAL: 2
    },
    sailthru: {
      getMessages: jest
        .fn()
        .mockResolvedValue([getNewOnMaxMessage_1(), getNewOnMaxMessage_4(), getNewOnMaxMessage_2(), getNewOnMaxMessage_3(), getSailthruMessage_1()]),
      markMessageAsRead: jest.fn().mockResolvedValue(true),
      getInitialNotification: jest.fn(),
      onNotificationTapped: jest.fn(cb => {
        cb({});
        return () => {};
      }) as any
    },
    reactNativePermission: require('react-native-permissions/mock'),
    geolocation: {
      addListener: jest.fn(),
      getCurrentPosition: jest.fn(),
      removeListeners: jest.fn(),
      requestAuthorization: jest.fn(),
      setConfiguration: jest.fn(),
      startObserving: jest.fn(),
      stopObserving: jest.fn()
    }
  } as any,
  stateSnapshot: {
    set: jest.fn(),
    get: jest.fn().mockReturnValue(initialStateGetter())
  },
  logger: {
    log: jest.fn(),
    info: jest.fn(),
    debug: jest.fn(),
    warn: jest.fn(),
    error: jest.fn(),
    success: jest.fn(),
    color: jest.fn(),
    assert: jest.fn(),
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    format: jest.fn()
  } as any,
  eventTrackerService: {
    trackDataUser: jest.fn(),
    isTrackingEnabled: jest.fn().mockResolvedValue(true),
    rZero: {
      trackView: jest.fn(),
      trackEvent: jest.fn(),
      flush: jest.fn(),
      setUser: jest.fn()
    },
    forterSDK: {
      setDevLogsEnabled: jest.fn(),
      getDeviceUniqueID: jest.fn(cb => cb()),
      init: jest.fn((a, b, c) => c()),
      trackNavigation: jest.fn(),
      trackNavigationWithExtraData: jest.fn(),
      trackAction: jest.fn(),
      trackActionWithJSON: jest.fn(),
      fraudTrackNavigationEvent: jest.fn(),
      fraudTrackAction: jest.fn()
    },
    tealiumSDK: {
      track: jest.fn(),
      trackView: jest.fn()
    },
    appsFlyerSDK: {
      updateServerUninstallToken: jest.fn(),
      setCustomerUserId: jest.fn().mockImplementation((_, cb) => cb('success')),
      addDeepLinkEventListener: jest.fn(),
      removeDeepLinkEventListener: jest.fn(),
      clearDeepLinkEventListener: jest.fn(),
      trackConversionEvent: jest.fn()
    }
  } as any,
  ccpaService: {
    showConsentBanner: jest.fn().mockResolvedValue(null),
    showPreferenceCenter: jest.fn()
  } as any,
  remoteConfigService: {
    getValue: jest.fn(async key => ENV.REMOTE_CONFIG.DEFAULT_CONFIG[key]),
    fetchAndActivate: jest.fn().mockResolvedValue(null)
  } as any,
  authService: {
    auth: {
      refreshToken: jest.fn().mockResolvedValue(true)
    },
    webAuth: {
      authorize: jest.fn(),
      clearSession: jest.fn()
    }
  } as any
});
