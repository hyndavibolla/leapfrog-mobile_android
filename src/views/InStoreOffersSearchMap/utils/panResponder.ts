const swipeConfig = {
  velocityThreshold: 0.3,
  directionalOffsetThreshold: 80,
  gestureIsClickThreshold: 5
};

const swipeDirections = {
  swipeUp: 'SWIPE_UP',
  swipeDown: 'SWIPE_DOWN'
};

export const handleShouldSetPanResponder = ({ nativeEvent: { touches } }, gestureState) => {
  return touches.length === 1 && !gestureIsClick(gestureState);
};

export const gestureIsClick = ({ dx, dy }) => {
  return Math.abs(dx) < swipeConfig.gestureIsClickThreshold && Math.abs(dy) < swipeConfig.gestureIsClickThreshold;
};

export const handlePanResponderEnd = (gestureState, onSwipeUp, onSwipeDown) => {
  const swipeDirection = getSwipeDirection(gestureState);
  triggerSwipeHandlers(swipeDirection, onSwipeUp, onSwipeDown);
};

export const triggerSwipeHandlers = (swipeDirection, onSwipeUp, onSwipeDown) => {
  const { swipeUp, swipeDown } = swipeDirections;
  switch (swipeDirection) {
    case swipeUp:
      onSwipeUp();
      break;
    case swipeDown:
      onSwipeDown();
      break;
  }
};

export const getSwipeDirection = gestureState => {
  const { swipeUp, swipeDown } = swipeDirections;
  if (isValidVerticalSwipe(gestureState)) {
    return gestureState.dy > 0 ? swipeDown : swipeUp;
  }
  return null;
};

export const isValidVerticalSwipe = ({ vy, dx }) => {
  const { velocityThreshold, directionalOffsetThreshold } = swipeConfig;
  return isValidSwipe(vy, velocityThreshold, dx, directionalOffsetThreshold);
};

export const isValidSwipe = (velocity, velocityThreshold, directionalOffset, directionalOffsetThreshold) => {
  return Math.abs(velocity) > velocityThreshold && Math.abs(directionalOffset) < directionalOffsetThreshold;
};
