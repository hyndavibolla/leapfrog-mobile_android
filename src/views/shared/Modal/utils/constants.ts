export enum ModalType {
  CENTER = 'center',
  BOTTOM = 'bottom'
}

export enum ModalSize {
  SMALL = 'small',
  MEDIUM = 'medium',
  EXTRA_LARGE = 'extralarge',
  DYNAMIC = 'dynamic',
  FULL_SCREEN = 'fullscreen'
}

export const ModalSizeToHeightMap = {
  [ModalSize.SMALL]: 40,
  [ModalSize.MEDIUM]: 50,
  [ModalSize.EXTRA_LARGE]: 80,
  [ModalSize.FULL_SCREEN]: 100
};
