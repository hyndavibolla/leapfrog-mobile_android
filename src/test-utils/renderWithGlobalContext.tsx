import React, { ReactElement } from 'react';
import { render } from '@testing-library/react-native';

import { getInitialState, GlobalProvider } from '../state-mgmt/GlobalState';
import { Deps, IGlobalState, IReducer } from '../models/general';
import { getMockDeps } from './getMockDeps';
import { getMockReducer } from './getMockReducer';

export const renderWithGlobalContext = (
  children: ReactElement,
  deps: Deps = getMockDeps(),
  initState: IGlobalState = getInitialState(),
  combinedReducers: IReducer<IGlobalState, any> = getMockReducer()
) => {
  const renderAPI = render(
    <GlobalProvider deps={deps} initState={initState} combinedReducers={combinedReducers}>
      {children}
    </GlobalProvider>
  );
  return { ...renderAPI, mockReducer: combinedReducers, deps };
};
