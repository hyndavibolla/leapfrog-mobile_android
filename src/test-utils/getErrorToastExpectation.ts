import { act } from 'react-test-renderer';

import { ACTION_TYPE } from '../state-mgmt/core/actions';
import { ToastType } from '../views/shared/Toast';
import { Deps } from '../models/general';
import { wait } from '../utils/wait';
import { renderWrappedHook } from './renderWrappedHook';

export const getErrorToastExpectation = async (hookGetter: () => any, deps: Deps, errorArgs: any[] = []) => {
  const { result, mockReducer } = renderWrappedHook(hookGetter, deps);
  await act(async () => {
    await (result.current[0] as (...args: any[]) => Promise<any>)(...errorArgs);
    await wait(0); // waiting for the useEffect dispatching the toast to run
    expect(mockReducer).toBeCalledWith(
      expect.any(Object),
      expect.objectContaining({
        type: ACTION_TYPE.ADD_TOAST,
        payload: expect.objectContaining({ props: expect.objectContaining({ type: ToastType.ERROR }) })
      })
    );
  });
};
