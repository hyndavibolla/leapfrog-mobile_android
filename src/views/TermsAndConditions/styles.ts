import { css } from '@emotion/native';
import { COLOR } from '../../constants';

export const styles = {
  spinnerContainer: css`
    position: absolute;
    flex: 1;
    width: 100%;
    height: 100%;
    top: 0;
    left: 0;
    justify-content: center;
    align-items: center;
    background-color: ${COLOR.LIGHT_GRAY};
  `
};
