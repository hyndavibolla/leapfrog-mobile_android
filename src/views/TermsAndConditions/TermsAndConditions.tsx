import React, { memo, useContext, useEffect, useMemo } from 'react';
import { View } from 'react-native';

import { WebView } from '../shared/WebView';
import { ConnectionBanner } from '../shared/ConnectionBanner';
import { Spinner } from '../shared/Spinner';
import { styles } from './styles';
import { GlobalContext } from '../../state-mgmt/GlobalState';
import { useAsyncCallback } from '../../utils/useAsyncCallback';
import { getWrappedHtml } from '../../utils/getWrappedStaticHtml';
import { CriticalError } from '../shared/CriticalError/CriticalError';
import { useTestingHelper } from '../../utils/useTestingHelper';

export const TermsAndConditions = () => {
  const { getTestIdProps } = useTestingHelper('terms-and-conditions');
  const { deps } = useContext(GlobalContext);
  const [onGetStaticHtml, isGetStaticHtmlLoading = true, error, staticHtml] = useAsyncCallback(async () => {
    const url = 'https://www.shopyourway.com/secured/-/m/static-content?type=TermsOfService';
    return await deps.httpService.fetch(url, { method: 'GET' }).then(r => r.text());
  }, []);

  useEffect(() => {
    onGetStaticHtml();
  }, [onGetStaticHtml]);

  const html = useMemo(() => getWrappedHtml(staticHtml), [staticHtml]);

  return (
    <>
      <ConnectionBanner />
      {!isGetStaticHtmlLoading ? null : (
        <View style={styles.spinnerContainer} {...getTestIdProps('spinner')}>
          <Spinner />
        </View>
      )}
      {!error ? null : <CriticalError />}
      {!staticHtml ? null : <WebView source={{ html }} />}
    </>
  );
};

export default memo(TermsAndConditions);
