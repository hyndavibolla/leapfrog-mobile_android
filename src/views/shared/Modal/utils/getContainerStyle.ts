import { memorize } from '../../../../utils/memorize';
import { styles } from '../styles';
import { ModalSize, ModalType, ModalSizeToHeightMap } from './constants';

interface props {
  height: number;
  modalHeight: number;
  size: ModalSize;
  type: ModalType;
}

const getContainerStyle = ({ height, modalHeight, size, type }: props) => {
  switch (type) {
    case ModalType.BOTTOM:
      if (size === ModalSize.FULL_SCREEN) return [styles.containerBottom, styles.containerBottomFullScreen];
      else if (size !== ModalSize.DYNAMIC) return [styles.containerBottom, { maxHeight: (ModalSizeToHeightMap[size] * height) / 100 }];
      return [styles.containerBottom];
    case ModalType.CENTER:
      if (size === ModalSize.DYNAMIC) return [styles.containerCenter, styles.containerCenterDynamic, { bottom: (height - modalHeight) / 2 }];
      const center = (ModalSizeToHeightMap[size] * height) / 100;
      return [styles.containerCenter, { height: center, bottom: (height - center) / 2 }];
  }
};

export default memorize(getContainerStyle);
