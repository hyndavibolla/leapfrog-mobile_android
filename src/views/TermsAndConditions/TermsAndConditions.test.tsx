import React from 'react';
import { act } from 'react-test-renderer';

import { TermsAndConditions } from './TermsAndConditions';
import { renderWithGlobalContext } from '../../test-utils/renderWithGlobalContext';
import { wait } from '../../utils/wait';
import { getMockDeps } from '../../test-utils/getMockDeps';

describe('TermsAndConditions', () => {
  it('should render loading', async () => {
    const deps = getMockDeps();
    let resolve;
    const pendingPromise = new Promise(r => {
      resolve = r;
    });
    deps.httpService.fetch = jest.fn().mockReturnValue(pendingPromise);
    const { toJSON } = renderWithGlobalContext(<TermsAndConditions />, deps);
    expect(toJSON()).toMatchSnapshot();
    resolve();
    await act(() => wait(0));
  });

  it('should render', async () => {
    const deps = getMockDeps();
    deps.httpService.fetch = jest.fn().mockResolvedValue({ text: () => Promise.resolve('Terms and conditions text') });
    const { toJSON } = renderWithGlobalContext(<TermsAndConditions />, deps);
    await act(() => wait(0));
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with error', async () => {
    const deps = getMockDeps();
    deps.httpService.fetch = Promise.reject;
    const { queryByTestId } = renderWithGlobalContext(<TermsAndConditions />, deps);
    await act(() => wait(0));
    expect(queryByTestId('critical-error-container')).toBeTruthy();
  });
});
