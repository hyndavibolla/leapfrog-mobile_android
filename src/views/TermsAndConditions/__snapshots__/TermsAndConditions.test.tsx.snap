// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TermsAndConditions should render 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "overflow": "hidden",
      },
      undefined,
    ]
  }
>
  <RNCWebView
    cacheEnabled={true}
    injectedJavaScript="true;"
    injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
    injectedJavaScriptForMainFrameOnly={true}
    javaScriptEnabled={true}
    messagingEnabled={true}
    onContentProcessDidTerminate={[Function]}
    onError={[Function]}
    onHttpError={[Function]}
    onLoad={[Function]}
    onLoadingError={[Function]}
    onLoadingFinish={[Function]}
    onLoadingProgress={[Function]}
    onLoadingStart={[Function]}
    onMessage={[Function]}
    onShouldStartLoadWithRequest={[Function]}
    setSupportMultipleWindows={false}
    sharedCookiesEnabled={true}
    source={
      Object {
        "html": "<html>
<head>
  <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
  <style>
    * { font-family: Gotham, \\"Helvetica Neue\\", Helvetica, Arial, \\"sans-serif\\";
        font-size: 12px;
        font-style:normal;
        text-align: justify; 
        line-height: 1.6em;
    }
    body { margin: 20px; }
    h1 { display: none }
    h2 {
      font-size: 25px;
      color: #517E99;
      font-weight: bold;
    }

    em {
      font-style: normal;
      font-weight: bold;
      font-size: 14px;
      color: #262C2F;
    }
    strong {
      font-style: normal;
      font-weight: bold;
      font-size: 14px;
      color: #262C2F;
    }
    a {
      font-size: 12px;
      font-style: normal;
      font-weight: 300;
      color: #262C2F;
    }

    tr td p {
      text-align: left;
    }

    td {
      padding: 10px
    }
    
  </style>
</head>
<body>Terms and conditions text</body></html>",
      }
    }
    startInLoadingState={false}
    style={
      Array [
        Object {
          "flex": 1,
          "overflow": "hidden",
        },
        Object {
          "backgroundColor": "#ffffff",
        },
        undefined,
      ]
    }
    textInteractionEnabled={true}
    thirdPartyCookiesEnabled={true}
    useSharedProcessPool={true}
  />
</View>
`;

exports[`TermsAndConditions should render loading 1`] = `
<View
  accessibilityLabel="terms-and-conditions-spinner"
  accessible={true}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5F7FA",
      "flexBasis": 0,
      "flexGrow": 1,
      "flexShrink": 1,
      "height": "100%",
      "justifyContent": "center",
      "left": 0,
      "position": "absolute",
      "top": 0,
      "width": "100%",
    }
  }
  testID="terms-and-conditions-spinner"
>
  <div>
    Spinner
  </div>
</View>
`;
